File: app/src/main/java/com/anokmik/tripassistant/validator/TextLengthValidator.java
package com.anokmik.tripassistant.validator;

import android.text.TextUtils;

public final class TextLengthValidator {

    public boolean higherThan(String text, @Threshold int threshold) {
        return notEmpty(text) && (text.length() > threshold);
    }

    public boolean lowerThan(String text, @Threshold int threshold) {
        return notEmpty(text) && (text.length() < threshold);
    }

    public boolean notEmpty(String text) {
        return !TextUtils.isEmpty(text);
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/validator/TextLengthValidator.kt
package com.anokmik.tripassistant.validator

import android.text.TextUtils

class TextLengthValidator {

    fun higherThan(text: String?, @Threshold threshold: Long) = notEmpty(text) && ((text?.length ?: 0) > threshold)

    fun lowerThan(text: String?, @Threshold threshold: Long) = notEmpty(text) && ((text?.length ?: 0) < threshold)

    fun notEmpty(text: String?) = !TextUtils.isEmpty(text)

}